Tests/SonarScanner.Shim.Tests/PropertiesFileGeneratorTests.cs

```diff
@@ -553,6 +553,28 @@ public void FileGen_AdditionalProperties()
             provider.AssertSettingDoesNotExist(SonarProperties.SonarPassword);
         }
 
+        [TestMethod]
+        public void FileGen_WhenNoGuid_LogWarnings()
+        {
+            // Arrange
+            var analysisRootDir = TestUtils.CreateTestSpecificFolder(TestContext);
+            var logger = new TestLogger();
+
+            CreateProjectWithFiles("project1", analysisRootDir, Guid.Empty);
+            var config = CreateValidConfig(analysisRootDir);
+
+            // Act
+            var result = new PropertiesFileGenerator(config, logger).GenerateFile();
+
+            // Assert
+            result.RanToCompletion.Should().BeFalse();
+            AssertExpectedProjectCount(1, result);
+            AssertFailedToCreatePropertiesFiles(result, logger);
+
+            logger.Warnings.Should().HaveCount(1);
+            logger.Warnings[0].Should().StartWith("The following projects don't have any project GUID and will be skipped from the analysis:");
```

Comment:
Should be "a", not "any".

Suggested slight rewording: "do not have a ProjectGuid..."